services/controller/standard/events.go

```diff
@@ -332,3 +339,61 @@ func (s *Service) subscribeToBeaconCommittees(ctx context.Context,
 	s.subscriptionInfos[epoch] = subscriptionInfo
 	s.subscriptionInfosMutex.Unlock()
 }
+
+// VerifySyncCommitteeMessages handles the "head" events from the beacon node for sync committee validation.
+func (s *Service) VerifySyncCommitteeMessages(ctx context.Context, data *apiv1.HeadEvent) {
+	_, span := otel.Tracer("attestantio.vouch.services.controller.standard").Start(ctx, "VerifySyncCommitteeMessages")
+	defer span.End()
+
+	// We verify against the previous slot as that is when the sync committee will have reported.
+	previousSlot := data.Slot - 1
+	currentSlot := data.Slot
+
+	// Logging with the current slot as that is when this code is executed.
+	log := log.With().
+		Uint64("current_slot", uint64(currentSlot)).
+		Uint64("previous_slot", uint64(previousSlot)).
+		Str("block_header", data.Block.String()).
+		Logger()
+	log.Trace().Msg("Received head event")
+
+	previousSlotData, found := s.syncCommitteeMessenger.GetDataUsedForSlot(previousSlot)
+	if !found {
+		log.Trace().Msg("No reported sync committee message data for slot; skipping validation")
```

Comment:
Any path that is unexpected (such as this one) but doesn't have an impact on the direct workings of the function we should log at `Debug` level.  This fits with the use of `Debug` level elsewhere.

This is true for a few other log messages in this function as well.

Reply:
I understand the point and have addressed for other lines, but I think we _do_ expect this log line frequently. That is because the slot data is only stored when we broadcast a sync committee message, so most of the time we would not find previous slot data.

Reply:
Ah right, so ideally we would want to only run this function if we knew we were participating in a sync committee, but that's going to be slightly harder on the logic side.  Let's leave this as-is for now, then.